server/src/main/java/org/elasticsearch/common/util/concurrent/ThreadContext.java

```diff
@@ -134,6 +135,61 @@ public StoredContext stashContext() {
         };
     }
 
+    /**
+     * Removes the specified transient headers from the current context. When the returned
+     * {@link StoredContext} is closed, it will restore these transient headers to their original
+     * value (including restoring them to an <i>unset</i> value if they did not originally exist).
+     * Closing the {@code StoredContext} has no affect on any other header - any headers
+     * (other than those names specified in {@code transientHeadersToStash} that were
+     * added to the {@code ThreadContext} will be retained.
+     *
+     * For example, at the end of the following code, the ThreadContext will have transient
+     * values {@code "a"=1}, {@code "b"=1}, {@code "d"=2} and {@code "c"} will not be set.
```

Comment:
```suggestion
     * values {@code "a"=1}, {@code "b"=1}, {@code "d"=2} and {@code "c"=2} will not be set.
```

Reply:
`c` not being set is the correct thing.
It is included in the list of headers to be stashed, but it is not part of the context before the stash. Therefore the behaviour after restore is to also not be present.